File: tests/adapters/java/goldens/literals/max_tokens_20.java
/**
 * Java module for testing literal optimization.
 */

package com.example.literals;

import java.util.*;
import java.time.Instant;

// Short string literal (should be preserved)
class Constants {
    public static final String SHORT_MESSAGE = "Hello, World!";

    // Long string literal (candidate for trimming)
    public static final String LONG_MESSAGE = "This is an extremely long message that contains a substantial amount of text content which might be considered…"; // literal string (−54 tokens)

    // Multi-line string with embedded expressions (Java 15+ text blocks)
    public static final String TEMPLATE_WITH_DATA = String.format("""
        User Information:
        - Name: %s…""", // literal string (−50 tokens)
        getUserName(),
        getUserEmail(),
        Instant.now().toString(),
        getAccountStatus(),
        String.join(", ", getPermissions()),
        getLastLogin(),
        getProfileCompleteness()
    );

    private static String getUserName() { return "John Doe"; }
    private static String getUserEmail() { return "john.doe@example.com"; }
    private static String getAccountStatus() { return "active"; }
    private static List<String> getPermissions() { return List.of("read", "write", "admin"); }
    private static String getLastLogin() { return "2024-01-15T10:30:00Z"; }
    private static int getProfileCompleteness() { return 85; }
}

class DataContainer {
    // Small array (should be preserved)
    private final List<String> tags;

    // Large array (candidate for trimming)
    private final List<String> items;

    // Small object (should be preserved)
    private final Map<String, Object> metadata;

    // Large object (candidate for trimming)
    private final Map<String, Object> configuration;

    public DataContainer(List<String> tags, List<String> items,
                        Map<String, Object> metadata, Map<String, Object> configuration) {
        this.tags = tags;
        this.items = items;
        this.metadata = metadata;
        this.configuration = configuration;
    }

    public List<String> getTags() { return tags; }
    public List<String> getItems() { return items; }
    public Map<String, Object> getMetadata() { return metadata; }
    public Map<String, Object> getConfiguration() { return configuration; }
}

public class LiteralDataManager {
    // Class properties with various literal types
    private final Map<String, Object> smallConfig = Map.of(
        "debug", true,
        "version", "1.0.0"
    );

    private final Map<String, Object> largeConfig = Map.ofEntries(
        Map.entry(
            "database",
            // … (1 more, −140 tokens)
        )
        // … (3 more, −379 tokens)
    );

    private final List<String> supportedLanguages;
    private final Set<String> allowedExtensions;

    public LiteralDataManager() {
        // Array with many elements (trimming candidate)
        this.supportedLanguages = List.of(
            "english",
            "spanish",
            "french"
            // … (21 more, −82 tokens)
        );

        // Set with many elements
        this.allowedExtensions = Set.of(
            ".java",
            ".kt",
            ".scala",
            ".groovy"
            // … (14 more, −43 tokens)
        );
    }

    public DataContainer processData() {
        // Function with various literal data
        List<String> smallArray = List.of("one", "two", "three");

        List<String> largeArray = List.of(
            "item_001",
            "item_002"
            // … (28 more, −140 tokens)
        );

        Map<String, Object> nestedData = Map.of(
            "level1", Map.of(
                "level2",
                // … (1 more, −188 tokens)
            )
        );

        return new DataContainer(
            smallArray,
            largeArray,
            Map.of("type", "test", "count", smallArray.size()),
            nestedData
        );
    }

    public String getLongQuery() {
        // Very long SQL-like query string
        return """
            SELECT
                users.id, users.username, users.email, users…"""; // literal string (−173 tokens)
    }

    public List<String> getSupportedLanguages() { return supportedLanguages; }
    public Set<String> getAllowedExtensions() { return allowedExtensions; }
}

// Module-level constants with different sizes
class SmallConstants {
    public static final Map<String, Object> VALUES = Map.of(
        "API_VERSION", "v1",
        "DEFAULT_LIMIT", 50
    );
}

class LargeConstants {
    public static final Map<String, Object> HTTP_STATUS_CODES = Map.ofEntries(
        Map.entry("CONTINUE", 100)
        // … (40 more, −401 tokens)
    );

    public static final Map<String, String> ERROR_MESSAGES = Map.ofEntries(
        Map.entry("VALIDATION_FAILED", "Input validation failed. Please check your data and try again.")
        // … (6 more, −110 tokens)
    );
}

File: tests/adapters/scala/goldens/literals/max_tokens_20.scala
/**
 * Scala module for testing literal optimization.
 */

package com.example.literals

// Short string literal (should be preserved)
object Constants {
  val SHORT_MESSAGE = "Hello, World!"

  // Long string literal (candidate for trimming)
  val LONG_MESSAGE = """This is an extremely long message that contains a substantial amount of text content which might be consi…""" // literal string (−54 tokens)

  // Multi-line string with interpolation
  def getUserName(): String = "John Doe"
  def getUserEmail(): String = "john.doe@example.com"
  def getAccountStatus(): String = "active"
  def getPermissions(): List[String] = List("read", "write", "admin")
  def getLastLogin(): String = "2024-01-15T10:30:00Z"
  def getProfileCompleteness(): Int = 85

  val TEMPLATE_WITH_DATA = s"""
User Information:
- Name: ${getUserName()}
- Email: ${getUserEmail()}…""" // literal string (−54 tokens)
}

case class DataContainer(
  // Small array (should be preserved)
  tags: List[String],

  // Large array (candidate for trimming)
  items: List[String],

  // Small object (should be preserved)
  metadata: Map[String, Any],

  // Large object (candidate for trimming)
  configuration: Map[String, Any]
)

class LiteralDataManager {
  // Class properties with various literal types
  private val smallConfig = Map(
    "debug" -> true,
    // … (1 more, −11 tokens)
  )

  private val largeConfig = Map(
    "database" -> Map(
        "host" -> "localhost",
        // … (5 more, −101 tokens)
    ),
    // … (3 more, −304 tokens)
  )

  private val supportedLanguages: List[String] = List(
    "english",
    "spanish",
    "french"
    // … (21 more, −82 tokens)
  )

  private val allowedExtensions: Set[String] = Set(
    ".scala",
    ".sc",
    ".java",
    ".kt"
    // … (14 more, −43 tokens)
  )

  def processData(): DataContainer = {
    // Function with various literal data
    val smallArray = List("one", "two", "three")

    val largeArray = List(
      "item_001",
      "item_002"
      // … (28 more, −140 tokens)
    )

    val nestedData = Map(
      "level1" -> Map(
          "level2" -> Map(
              "level3" -> Map(
                  "data" -> List(
                      Map("id" -> 1, "name" -> "First", "active" -> true),
                      // … (4 more, −85 tokens)
                  ),
                  // … (1 more, −137 tokens)
              ),
          ),
      ),
    )

    DataContainer(
      tags = smallArray,
      items = largeArray,
      metadata = Map("type" -> "test", "count" -> smallArray.size),
      configuration = nestedData
    )
  }

  def getLongQuery(): String = {
    // Very long SQL-like query string
    """
      SELECT
        users.id, users.username, users.email, users.cre…""" /* literal string (−171 tokens) */.stripMargin
  }
}

// Module-level constants with different sizes
object SmallConstants {
  val VALUES = Map(
    "API_VERSION" -> "v1",
    "DEFAULT_LIMIT" -> 50
  )
}

object LargeConstants {
  val HTTP_STATUS_CODES = Map(
    "CONTINUE" -> 100,
    // … (40 more, −319 tokens)
  )

  val ERROR_MESSAGES = Map(
    "VALIDATION_FAILED" -> "Input validation failed. Please check your data and try again.",
    // … (6 more, −104 tokens)
  )
}
